### PR TITLE
reconcile: parallel --setup wizard merge + close H-1/H-2 secret leaks

### PR DESCRIPTION
**Pull request by SchwartzKamel**
## Summary

Reconciles parallel `--setup` wizard implementations (local stash + upstream PR #14) and closes two High-severity secret-disclosure surfaces flagged by Newman's audit.

Three commits on this branch:

| SHA | Commit |
|-----|--------|
| `85a59bf` | `fix(setup): reconcile parallel wizard implementations` |
| `8cfb9bf` | `feat(setup): auto-launch wizard on bare az-ai with no creds; document --init-wizard alias` |
| `c138c75` | `fix(security): close H-1 + H-2 secret-disclosure surfaces in wizard` |

Diff vs `main`: +751 / −124 across 6 files.

## Background

Local work-in-progress (preserved as `stash@{0}`, dated 2026-04-22) was independently building the same `--setup` wizard upstream just shipped in PR #14. The stash version had stronger security/UX invariants (TTY guard, masked `*` input, exit 130 on Ctrl+C, 0600 perms via `UserConfig.Save`); upstream had bare-subcommand parser work (`help` / `setup` literal-arg handling) and `SetupAndBareSubcommandTests.cs`. Neither version dominated the other — this branch combines both.

### Commit 1 — `85a59bf`: reconcile

- Rewrote `SetupWizard.cs` (354 lines) with upstream's `RunAsync()` signature + stash's invariants:
  - `IsInteractiveTty()` guard (refuses redirected stdin/stdout)
  - Masked API-key input via `Console.ReadKey(intercept: true)`, `*` echo, Backspace + Esc + EOF handling
  - Success-only `UserConfig.Save` (no partial writes on Ctrl+C)
- `Program.cs` `--setup` branch refuses `--raw` / `--json` / non-TTY before invoking the wizard
- `UserConfig.cs` adds `Endpoint` / `ApiKey` JSON fields, `SetKey` / `GetKey` cases for `endpoint` / `api_key`, `ListKeys` redaction
- `SetupWizardTests.cs` ports stash's parser-flag + UserConfig round-trip + redaction tests (108 lines, no overlap with upstream's `SetupAndBareSubcommandTests.cs`)

### Commit 2 — `8cfb9bf`: auto-launch + alias docs

- Implemented bare-`az-ai`-with-no-creds wizard auto-trigger via pure `ShouldAutoLaunchSetup` predicate (6 unit tests, no PTY harness needed)
- Env → `UserConfig` credential fallback so persisted-config users don't get the wizard mis-fired
- Documented `--init-wizard` alias in `ShowHelp()`, embedded bash / zsh / fish completion strings
- Closed CHANGELOG-vs-code drift (the `[Unreleased]` entry already promised this behavior)

### Commit 3 — `c138c75`: H-1 + H-2 hardening

Closes Newman's two High-severity findings:

- **H-1**: `ReadMaskedLine` no longer falls back to unmasked `Console.ReadLine` when `ReadKey` throws on broken pseudo-TTYs (containers, certain CI runners, WSL+ssh edge cases). Now fails closed with stderr warning + exit 130. Static-analysis regression test guards against future "let's just gracefully fall back" patches.
- **H-2**: `--config get api_key` now refuses with a helpful message instead of printing the raw key to stdout. JSON-mode-aware via `ErrorAndExit`. `UserConfig.GetKey("api_key")` itself still returns the raw value for in-process callers (the wizard needs it); the fix lives at the print site.

## Test impact

- **Unit tests**: 581 passing (was 567 on `main`, +14 new)
- **Integration tests**: 34 passing, 2 skipped (require live Azure creds)
- **Preflight**: green at every commit (format, color-contract-lint, build, unit, integration)

## Backlog (deferred to separate PRs)

Newman's Medium findings — none of these block this PR:

- **M-1**: TOCTOU window between `File.WriteAllText` and `SetUnixFileMode` in `UserConfig.Save` (atomic-rename pattern needed)
- **M-2**: SSRF / private-IP warning at endpoint config time (share helper with `WebFetchTool` allowlist)
- **M-3**: Windows DACL not enforced on persisted JSON
- **M-5**: 16-char API key min-length is a soft floor that could reject future Azure key formats
- (M-4 was subsumed by H-1 fix.)

## Reviewers

- **Newman**: H-1 / H-2 closure verification (audit was on `85a59bf`; commit 3 closes both findings)
- **Mr. Lippman**: CHANGELOG `[Unreleased]` entries are now consistent with shipped behavior
- **Bob Sacamano**: completion scripts now expose `--init-wizard`; Homebrew / Scoop / Nix consumers regenerating completions pick it up automatically
- **Elaine** *(post-merge)*: `man/az-ai.1` predates the wizard entirely — separate PR to add the Setup section

Co-authored-by: Copilot <223556219+Copilot@users.noreply.github.com>

### CODE REVIEW

File: azureopenai-cli/SetupWizard.cs [hunk 1]
```diff
@@ -1,170 +1,290 @@
...
+                continue;
             }
-        }
 
-        // ── Step 4: Model deployment ──────────────────────────────────
-        var currentModel = Environment.GetEnvironmentVariable("AZUREOPENAIMODEL");
-        if (!string.IsNullOrWhiteSpace(currentModel))
-            Console.WriteLine($"  Current model: {currentModel}");
+            return input.TrimEnd('/');
+        }
```
[the remark this answers is not in the document]
Comment (SchwartzKamel) [reply azureopenai-cli/SetupWizard.cs]: @copilot apply changes based on [this feedback](https://github.com/SchwartzKamel/azure-openai-cli/pull/15#discussion_r3157030139)

File: azureopenai-cli/Program.cs [hunk 2]
```diff
@@ -1895,6 +1971,22 @@ internal static int HandleConfigSubcommand(CliOptions opts, UserConfig config)
                 {
                     return ErrorAndExit("--config get requires <key>", 1, opts.Json);
                 }
```
[the remark this answers is not in the document]
Comment (SchwartzKamel) [reply azureopenai-cli/Program.cs:1973]: @copilot apply changes based on [this feedback](https://github.com/SchwartzKamel/azure-openai-cli/pull/15#discussion_r3157030158)

File: tests/AzureOpenAI_CLI.Tests/SetupWizardTests.cs [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+    public void UserConfig_EndpointAndApiKey_RoundTrip()
+    {
+        var path = Path.Combine(Path.GetTempPath(), $"az-ai-wizard-rt-{Guid.NewGuid():N}.json");
+        try
+        {
+            var cfg = new UserConfig
+            {
+                Endpoint = "https://example.openai.azure.com",
+                ApiKey = "sk-test-0123456789abcdef-0123456789abcdef",
+                DefaultModel = "default",
```
[the remark this answers is not in the document]
Comment (SchwartzKamel) [reply tests/AzureOpenAI_CLI.Tests/SetupWizardTests.cs:54]: @copilot apply changes based on [this feedback](https://github.com/SchwartzKamel/azure-openai-cli/pull/15#discussion_r3157030119)
